packages/core/src/common/keyboard/layout-provider.ts

```diff
@@ -0,0 +1,37 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { IKeyboardLayoutInfo, IKeyboardMapping } from 'native-keymap';
+import { Event } from '../event';
+
+export const keyboardPath = '/services/keyboard';
+
+export const KeyboardLayoutProvider = Symbol('KeyboardLayoutProvider');
+
+export interface KeyboardLayoutProvider {
+    getNativeLayout(): Promise<NativeKeyboardLayout>;
+}
+
+export const KeyboardLayoutChangeNotifier = Symbol('KeyboardLayoutChangeNotifier');
+
+export interface KeyboardLayoutChangeNotifier {
+    onNativeLayoutChanged: Event<NativeKeyboardLayout>;
```

Comment:
Is it fired then a user change keyboard layout? I'm testing changing from EN to DE and back, and it does not work for me, i had to reload Chrome. (on mac)

Reply:
looks like `layoutchange` event is not implemented on mac :(

Reply:
I am also not seeing layout updates at runtime when running Electron on Linux.

But the event doesn't seem to be implemented on Linux at all, so it seems normal:
https://github.com/Microsoft/node-native-keymap/blob/master/src/keyboard_x.cc#L235-L237

Reply:
https://github.com/Microsoft/vscode/issues/17521#issuecomment-267933654

> we now detect keyboard layout changes on OSX, no reasonable solution was found on Linux.